test/integration/huge_payload_test.rb

```diff
@@ -23,7 +23,7 @@ def setup
   end
 
   def test_report_handles_huge_payloads
-    payload = generate_payload((ENV['HUGE_PAYLOAD_SIZE'] || 10000).to_i)
+    payload = generate_payload((ENV['HUGE_PAYLOAD_SIZE'] || 2000).to_i)
```

Comment:
I had to reduce this number. This PR bumps rack to `3.1.14` which solves [CVE-2025-46727](https://access.redhat.com/security/cve/cve-2025-46727). The test was sending `20001` parameters but now only up to `4096` are permitted. Is it expected to receive more than `4096` parameters? because that scenario will fail from now on.

Do you know of any scenario where such amount of parameters can be received?

Reply:
Good question. Maybe under high load. Why was it previously set to 10k? :thinking:

Reply:
> Why was it previously set to 10k? 

I don't think there's a particular reason, just a high number probably.

> Maybe under high load

It's not directly related. Here we are not talking about 10000 requests, but one single request, with 20001 parameters in the URL. It's hard to think on a scenario when that can happen.

According to [docs](https://github.com/3scale/porta/blob/5a062aaf2ce3620a2969d0a7c50ce55283310c51/doc/active_docs/service_management_api.json#L388), only 1000 transactions per request are supported by this endpoint, that would require 2001 parameters in the URL. After the CVE fix, we accept up to 4096 parameters in the URL so I think we are fine.

Reply:
I'd like us to see what production traffic we see and to make sure we have a good margin over it.

But to save me from looking, could you tell me whether this is query parameters or POST data?

Reply:
POST data

Reply:
Check the `generate_payload` method, the received parameter is the number of parameters, but the total length must to be over 4096